src/codegen/generate-classes.ts

```diff
@@ -1435,6 +1449,7 @@ function generateZig(
     construct,
     finalize,
     noConstructor = false,
+    overridesToJS = false,
     estimatedSize,
```

Comment:
@dylan-conway What's this bit about? (I'm curious from a how we are generating classes and whether we can leverage extending from base classes or not)

Reply:
check where it is used, you'll find it determines if we generate `pub fn toJS`

this function is the convention for converting something to a JSValue. some functions check if the passed type has this method declared and calls it automatically in some places.

in general the classes generator is done with 
1. a c++ class that hooks into all the webkit apis, and fwiw *is* the class.
2. a zig struct that is constructed along with the c++ class `<1>` contains a pointer to `<2>`
3. a second zig struct which is `usingnamespace`d into `<2>` which provides a default set of methods that let you call back into webkit

Reply:
Yes what Dave said. In this situation it was useful to be able to override toJS to add debug assertions and to ensure reported_estimated_size is set

Reply:
Is there a way to check this in another place for the existence of method to avoid added metadata and allow overriding other things if specified like overriding `fromJS` or `getConstructor`.

Reply:
To do that we could parse the zig struct implementation for the existence of the methods (`toJS`, `fromJS`, ...), and if any are defined, skip it in the generated code. It's possible but probably too much work for now